Index: gvfs-1.48.1.2/client/gdaemonvfs.c
===================================================================
--- gvfs-1.48.1.2.orig/client/gdaemonvfs.c
+++ gvfs-1.48.1.2/client/gdaemonvfs.c
@@ -308,7 +308,7 @@ g_daemon_vfs_init (GDaemonVfs *vfs)
   vfs->wrapped_vfs = g_vfs_get_local ();
 
   /* Use the old .gvfs location as fallback, not .cache/gvfs */
-  if (g_get_user_runtime_dir() == g_get_user_cache_dir ())
+  if (g_strcmp0 (g_get_user_runtime_dir(), g_get_user_cache_dir ()) == 0)
     file = g_build_filename (g_get_home_dir(), ".gvfs", NULL);
   else
     file = g_build_filename (g_get_user_runtime_dir(), "gvfs", NULL);
Index: gvfs-1.48.1.2/daemon/main.c
===================================================================
--- gvfs-1.48.1.2.orig/daemon/main.c
+++ gvfs-1.48.1.2/daemon/main.c
@@ -79,7 +79,7 @@ on_name_acquired (GDBusConnection *conne
       char *argv2[6];
       
       /* Use the old .gvfs location as fallback, not .cache/gvfs */
-      if (g_get_user_runtime_dir() == g_get_user_cache_dir ())
+      if (g_strcmp0 (g_get_user_runtime_dir(), g_get_user_cache_dir ()) == 0)
         fuse_path = g_build_filename (g_get_home_dir(), ".gvfs", NULL);
       else
         fuse_path = g_build_filename (g_get_user_runtime_dir (), "gvfs", NULL);
Index: gvfs-1.48.1.2/daemon/mount.c
===================================================================
--- gvfs-1.48.1.2.orig/daemon/mount.c
+++ gvfs-1.48.1.2/daemon/mount.c
@@ -739,7 +739,7 @@ handle_register_mount (GVfsDBusMountTrac
       if (arg_user_visible)
         {
           /* Use the old .gvfs location as fallback, not .cache/gvfs */
-          if (g_get_user_runtime_dir() == g_get_user_cache_dir ())
+          if (g_strcmp0 (g_get_user_runtime_dir(), g_get_user_cache_dir ()) == 0)
             mount->fuse_mountpoint = g_build_filename (g_get_home_dir(), ".gvfs", mount->stable_name, NULL);
           else
             mount->fuse_mountpoint = g_build_filename (g_get_user_runtime_dir(), "gvfs", mount->stable_name, NULL);
